WordPress/Classes/ViewRelated/Notifications/Tools/NotificationCommentDetailCoordinator.swift

```diff
@@ -110,14 +110,39 @@ private extension NotificationCommentDetailCoordinator {
     }
 
     func updateViewWith(notification: Notification) {
-        if notification.kind == .comment {
-            trackDetailsOpened(for: notification)
-            onSelectedNoteChange?(notification)
-            configure(with: notification)
-            refreshViewController()
-        } else {
-            // TODO: handle other notification type
+        trackDetailsOpened(for: notification)
+        onSelectedNoteChange?(notification)
+
+        guard notification.kind == .comment else {
+            showNotificationDetails(with: notification)
+            return
         }
+
+        configure(with: notification)
+        refreshViewController()
+    }
+
+    func showNotificationDetails(with notification: Notification) {
+        let storyboard = UIStoryboard(name: Notifications.storyboardName, bundle: nil)
+
+        guard let viewController = viewController,
+        let notificationDetailsViewController = storyboard.instantiateViewController(withIdentifier: Notifications.viewControllerName) as? NotificationDetailsViewController else {
+            DDLogError("NotificationCommentDetailCoordinator: missing view controller.")
+            return
+        }
+
+        notificationDetailsViewController.note = notification
+        notificationDetailsViewController.notificationCommentDetailCoordinator = self
+        notificationDetailsViewController.dataSource = notificationsNavigationDataSource
+        notificationDetailsViewController.onSelectedNoteChange = onSelectedNoteChange
+
+        weak var navigationController = viewController.navigationController
+
+        viewController.dismiss(animated: true, completion: {
+            notificationDetailsViewController.navigationItem.largeTitleDisplayMode = .never
+            navigationController?.popViewController(animated: false)
```

Comment:
Just realized that with the `viewController.dismiss` and this line, this looks like the view controller is popped twice. But then I tested, and it just works 🌈 😄.

Reply:
I know, it _feels_ wrong. 😄  But yea that's the only way it works, both calls are necessary. 🤷